sensing_launch/launch/velodyne_node_container.launch.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2020 Tier IV, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import launch
+from launch import LaunchContext
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration, PythonExpression
+from launch_ros.actions import ComposableNodeContainer, LoadComposableNodes
+from launch_ros.descriptions import ComposableNode
+import uuid
+
+
+def acceptable_unique_name(prefix):
+    id = str(uuid.uuid4())
+    # ros2 apparently doesn't accept the UUID with hyphens in node names
+    return prefix + id.replace('-', '_')
+
+
+def generate_launch_description():
+    launch_arguments = []
+
+    def add_launch_arg(name: str, default_value=None):
+        # a default_value of None is equivalent to not passing that kwarg at all
+        launch_arguments.append(DeclareLaunchArgument(name, default_value=default_value))
+
+    add_launch_arg('model')
+    add_launch_arg('launch_driver', 'True')
+    add_launch_arg('calibration')
+    add_launch_arg('device_ip', '192.168.1.201')
+    add_launch_arg('sensor_frame', 'velodyne')
+    add_launch_arg('base_frame', 'base_link')
+    add_launch_arg('container_name', 'velodyne_composable_node_container')
+    add_launch_arg('min_range')
+    add_launch_arg('max_range')
+    add_launch_arg('pcap', '')
+    add_launch_arg('port', '2368')
+    add_launch_arg('read_fast', 'False')
+    add_launch_arg('read_once', 'False')
+    add_launch_arg('repeat_delay', '0.0')
+    add_launch_arg('rpm', '600.0')
+    add_launch_arg('laserscan_ring', '-1')
+    add_launch_arg('laserscan_resolution', '0.007')
+    add_launch_arg('num_points_thresholds', '300')
+    add_launch_arg('invalid_intensity')
+
+    def create_parameter_dict(*args):
+        result = {}
+        for x in args:
+            result[x] = LaunchConfiguration(x)
+        return result
+
+    nodes = []
+
+    # turn packets into pointcloud as in
+    # https://github.com/ros-drivers/velodyne/blob/ros2/velodyne_pointcloud/launch/velodyne_convert_node-VLP16-composed-launch.py
+    nodes.append(ComposableNode(
+        package='velodyne_pointcloud',
+        plugin='velodyne_pointcloud::Convert',
+        name='velodyne_convert_node',
+        parameters=[create_parameter_dict('velodyne_points', 'velodyne_points_ex', 'calibration',
```

Comment:
This shouldn't ask for velodyne_points or velodyne_points_ex since they are remapping topic values and not parameters.

```suggestion
        parameters=[create_parameter_dict('calibration',
```

Reply:
totally agree. fixed